lighthouse-cli/performance-experiment/server.js

```diff
@@ -84,18 +86,23 @@ function requestHandler(request, response) {
 }
 
 function reportRequestHandler(request, response) {
-  const html = reportGenerator.generateHTML(lhResults, 'cli');
+  const html = new ReportGenerator().generateHTML(lhResults, 'perf-x');
   response.writeHead(200, {'Content-Type': 'text/html'});
   response.end(html);
 }
 
+function blockedUrlPatternsRequestHandler(request, response) {
+  response.writeHead(200, {'Content-Type': 'text/json'});
+  response.end(JSON.stringify(lhParams.flags.blockedUrlPatterns || []));
+}
+
 function rerunRequestHandler(request, response) {
   let message = '';
   request.on('data', data => message += data);
 
   request.on('end', () => {
     const additionalFlags = JSON.parse(message);
-    const flags = Object.assign({}, lhParams.flags, additionalFlags);
+    const flags = Object.assign(lhParams.flags, additionalFlags);
```

Comment:
Because report page currently needs to ask perf-x server which URL patterns are blocked. So I need to store the blocked URL patterns somewhere.  Since in this PR we only store data for one report (including config, flags and results), I decided to directly store it with the flags (AFAIK there is no side effects).  In the future PR (#1477), flags will be stored on hard drive (for each experiment) and re-parsed when we need them.